upgrade/upgrade-scripts/upgrade-container

```diff
@@ -505,6 +505,63 @@ function migrate_configuration() {
 	EOF
 }
 
+function do_upgrade_container_in_place() {
+	run "$IMAGE_PATH/execute" -p "$(get_platform)" ||
+		die "'$IMAGE_PATH/execute' failed in '$CONTAINER'"
+
+	run /bin/systemctl reload delphix-platform ||
+		die "'systemctl reload delphix-platform' failed in '$CONTAINER'"
+
+	run /bin/systemctl restart delphix-platform ||
+		die "'systemctl restart delphix-platform' failed in '$CONTAINER'"
+}
+
+function do_upgrade_container_not_in_place() {
+	run "$IMAGE_PATH/execute" -p "$(get_platform)" ||
+		die "'$IMAGE_PATH/execute' failed in '$CONTAINER'"
+
+	run /bin/systemctl start delphix-platform ||
+		die "'systemctl start delphix-platform' failed in '$CONTAINER'"
+
+	migrate_configuration ||
+		die "failed to migrate configuration for '$CONTAINER'"
+}
+
+function get_type() {
+	local root origin
+
+	root=$(zfs list -Hpo name /var/lib/machines/"${CONTAINER}")
+	[[ -n "${root}" ]] || die "Failed to obtain root filesystem for container '${CONTAINER}'"
+	origin=$(zfs get -Hpo value origin "${root}")
```

Comment:
let's also add a `[[ -n "$origin" ]] || die ...` line after this, to account for the possibility that `zfs get` fails